src/eureka/S3_data_reduction/s3_reduce.py

```diff
@@ -293,6 +291,19 @@ def reduce(eventlabel, ecf_path=None, s2_meta=None, input_meta=None):
                 else:
                     meta.int_end = meta.int_start+meta.nplots
 
+                # Perform BG subtraction along dispersion direction
+                # for untrimmed NIRCam spectroscopic data
+                if hasattr(meta, 'bg_disp') and meta.bg_disp:
+                    meta.bg_dir = 'RxR'
+                    # Create bad pixel mask (1 = good, 0 = bad)
+                    data['mask'] = (['time', 'y', 'x'],
+                                    np.ones(data.flux.shape, dtype=bool))
+                    data = bg.BGsubtraction(data, meta, log,
+                                            m, meta.isplots_S3)
+                    meta.bg_disp = False
+                    meta.bg_deg = None
+                meta.bg_dir = 'CxC'
```

Comment:
Why set this to CxC afterwards? Don't you want this to be in an `else` statement?

Reply:
I think this at least needs some comment to help us remember what's going on here 6 months from now